### ISSUE TITLE
Todo list before the development branch can be moved to the master branch

### ISSUE THREAD

**Issue by ChristianFeldmann** [IENT/YUView#68]
I thought we should move towards getting the development branch to a state that we can move it to the main branch. Here we can collect what is still to be done and when it is, what should be tested before moving the branch. I think this also means that for now we should not try to add more features. Please feel free to add to this list.

- [x] Clean up caching threads and how this is handeled
- [x] Implement a caching strategy for the playback case

When these things are done, perform thorough testing:
- [x] playback/seeking 
- [x] loading of YUV/RGB sequences
- [x] containers: overlays/differences
- [x] playlists saving/loading
- [x] split view 
- [x] statistics
- [x] HEVC streams
- [x] keyboad shortcuts

**Comment by KubaO** [comment]
I'll work on caching threads. At the very least, stopping and restarting the caching threads on every request is very wasteful. A `QThread` is really a misnomer: it's a `QThreadHandle`, but they named it wrong. A finished thread doesn't persist in any way: the operating system disposes of it. Each and every time a thread is started, the underlying OS creates a new thread from scratch.

Thus, as a minimum fix we can use a `QRunnable` derived from `QObject` for signal support and submit it to the thread pool for execution. The threads will be started and stopped as needed.

I'll get that done in a few days.

**Comment by KubaO** [comment]
I can also deal with the playback caching, but I don't have any clear idea yet of how to implement it in a clean and understandable way. I don't particularly like the existing logic as it's hard to follow, but unless I can come up with something way better I'll probably leave it be until inspiration strikes :)

**Comment by ChristianFeldmann** [comment]
Ah okay so even though I create a list of QThreads which I 'reuse', the threads are still disposed of and recreated by the operation system. Because that is what I wanted to prevent by this approach. Mission not accomplished. I am quite new to the programming with QThreads. I understand the basic concepts but of course that's not all of it so any help is very much appreciated :)

**Comment by ChristianFeldmann** [comment]
I think this all woks. Let's see how many bugs will be found in the new version.
